src/components/helpers/PrivateRoute.tsx

```diff
@@ -0,0 +1,32 @@
+import { FC, useContext } from 'react';
+import { Redirect, Route } from 'react-router-dom';
+import { AuthContext } from '../../context/auth';
+
+interface PrivateRouteProps {
+  component: FC<any>;
+  exact?: boolean;
+  redirect?: string;
+  path: string;
+}
+
+export default function PrivateRoute({
+  component: Component,
+  exact = false,
+  redirect = '',
+  ...rest
+}: PrivateRouteProps) {
+  const { user } = useContext(AuthContext);
+  return redirect ? (
+    <Route
+      {...rest}
+      exact={!!exact}
+      render={(props) => (user ? <Redirect to={redirect} /> : <Component {...props} />)}
+    />
+  ) : (
+    <Route
+      {...rest}
+      exact={!!exact}
+      render={(props) => (user ? <Component {...props} /> : <Redirect to="/auth" />)}
```

Comment:
Hm, jedyną różnicą między tymi dwoma komponentami jest Redirect, czy idzie do auth czy do zmiennej redirect. Tylko trochę ciężko to zauważyć, bo ten cały return zawiera dwa warunki ternary. Na moje powinieneś rozbić go na dwa returny, a potem się zastanowić, czy dałoby się sensownie zbudować z nich jednego większego returna.

Reply:
Właśnie zastanawiałem się czy mozna to jakoś lepiej zrobić ale nie wpadłem na żaden pomysł